docs/source/user/threading-layer.rst

```diff
@@ -0,0 +1,119 @@
+.. _numba-threading-layer:
+
+The Numba threading layer
+=========================
+
+This section is about the Numba threading layer, this is the library that is
+used internally to perform the parallel execution that occurs through the use of
+the ``parallel`` targets for CPUs, namely:
+
+* The use of the ``parallel=True`` kwarg in ``@jit`` and ``@njit``.
+* The use of the ``target='parallel'`` kwarg in ``@vectorize`` and
+  ``@guvectorize``.
+
+.. note::
+    If a code base does not use the ``threading`` or ``multiprocessing``
+    modules (or any other sort of parallelism) the defaults for the threading
+    layer that ship with Numba will work well, no further action is required!
+
+
+Which threading layers are available?
+-------------------------------------
+There are three threading layers available and they are named as follows:
+
+* ``tbb`` - A threading layer backed by Intel TBB.
+* ``omp`` - A threading layer backed by OpenMP.
+* ``workqueue`` - A threading layer backed by OS native threads.
```

Comment:
well.. it's not like TBB and OMP are using some magic non-native threads ;-) Probably, `Simple work-sharing task scheduler`

Reply:
I was hoping to emphasise that there is no magic, it's just Numba using OS native threads with no 3rd party library involved, "built-in" does this fine, will change.